### PR TITLE
feat(launch): init support of launching applications

### PR DESCRIPTION
**Pull request by tai271828**
In the [sciwork online sprint today](https://hackmd.io/UbOReM5CQcmtP5gL1zxqqw?view), this is the feature we have implemented for issue: #26 .

# Steps to Verify the Pull Request
1. `source scripts/init`
1. `devenv use flavor-issue-26`
1. `devenv launch solvcon`

# Expected Result
1. SOLVCON will be built
2. `flavor/flavor-issue-26/application-solvcon` is created

# Additioinaly Information
This pull request only closes https://github.com/solvcon/devenv/issues/26 but not fully completes https://github.com/solvcon/devenv/discussions/23 for the follow incomplete tasks and imperfect features:

1. Uses still need to manually re-launch SOLVCON
2. SOLVCON source and lib are not fully integrated with `flavor/usr/`, `flavor/usr/bin`, and `flavor/src`.

I will create follow-up pull requests to complete above tasks later on to prevent a single huge pull request for code review. Inputs are welcome.

### CODE REVIEW

File: applications/solvcon/prepare-solvcon-dev.sh [hunk 1]
```diff
@@ -78,3 +85,11 @@ pushd ${SCSRC}/sandbox/gas/tube
 ./go run
 popd
 
+
```
Comment (tai271828) [inline applications/solvcon/prepare-solvcon-dev.sh:88]: This workaround will be removed when the whole story of integration between `devenv` and `SOLVCON` is done.

**Comment by tai271828** [comment]
The core part of this feature is currently referred to the implementation of `devenv build`. We will trace the architecture with the discussion thread https://github.com/solvcon/devenv/discussions/23 .

File: applications/solvcon/prepare-solvcon-dev.sh (hunk 1 above)
Comment (yungyuc) [reply applications/solvcon/prepare-solvcon-dev.sh:88]: What is the time / schedule of the whole integration?  Is the workaround intended to be checked in?

File: applications/solvcon/prepare-solvcon-dev.sh [hunk 2]
```diff
@@ -6,7 +6,14 @@
 #   source <this script> <your-project-folder>
 #
 set -x
-SOLVCON_PROJECT=${1:-${HOME}/solvcon}
+
+if [ -z "${DEVENVFLAVOR}" ]
+then
+  SOLVCON_PROJECT=${1:-${HOME}/solvcon}
+else
+  SOLVCON_PROJECT=${1:-${DEVENVAPPLICATION}}
```
Comment (yungyuc) [inline applications/solvcon/prepare-solvcon-dev.sh:14]: I think `DEVENVAPPLICATION` is unnecessarily long.  How about `DEVENVAPP`?

File: scripts/func.d/bash_utils [hunk 3]
```diff
@@ -18,6 +18,9 @@ get_list() {
     build)
       dir="${DEVENVROOT}/scripts/build.d"
       ;;
+    applicatioin)
```
Comment (yungyuc) [inline scripts/func.d/bash_utils:21]: "application"

**Review by yungyuc — CHANGES_REQUESTED** [review]

File: applications/solvcon/prepare-solvcon-dev.sh (hunk 2 above)
Comment (tai271828) [reply applications/solvcon/prepare-solvcon-dev.sh:14]: Excellent name!

File: applications/solvcon/prepare-solvcon-dev.sh (hunk 1 above)
Comment (tai271828) [reply applications/solvcon/prepare-solvcon-dev.sh:88]: I could not tell the E.T.A. right now. Let's evaluate the schedule by:
1. Checking how many packages are needed for SOLVCON
2. Checking how many packages are able to be built by `devenv`

In my pull request, `gmsh` has been integrated. I believe most packages could be integrated in the same way. So a rough schedule may be:
1. 2 man-days to complete above package enumeration
2. 2 man-days implementation mocking how `gmsh` is integrated
3. 2 man-days testing

Maybe 1 man-week?

**Comment by tai271828** [comment]
The updated PR is ready to review again. Thanks, team!

File: applications/solvcon/prepare-solvcon-dev.sh (hunk 1 above)
Comment (tai271828) [reply applications/solvcon/prepare-solvcon-dev.sh:88]: `shunit2` looks promising to me https://github.com/kward/shunit2/ . I will have a look of it in more details.

Comment (yungyuc) [reply applications/solvcon/prepare-solvcon-dev.sh:88]: > I could not tell the E.T.A. right now. Let's evaluate the schedule by:
> 
> 1. Checking how many packages are needed for SOLVCON
> 2. Checking how many packages are able to be built by `devenv`
> 
> In my pull request, `gmsh` has been integrated. I believe most packages could be integrated in the same way. So a rough schedule may be:
> 
> 1. 2 man-days to complete above package enumeration
> 2. 2 man-days implementation mocking how `gmsh` is integrated
> 3. 2 man-days testing
> 
> Maybe 1 man-week?

A rough estimate like weeks or months suffices for me.

**Review by yungyuc — APPROVED** [review]

**Pull request merged**